src/test/compile-fail/associated-types/cache/project-fn-ret-invariant.rs

```diff
@@ -60,9 +60,10 @@ fn baz<'a,'b>(x: Type<'a>) -> Type<'static> {
 
 #[cfg(krisskross)] // two instantiations, mixing and matching: BAD
 fn transmute<'a,'b>(x: Type<'a>, y: Type<'b>) -> (Type<'a>, Type<'b>) {
-   let a = bar(foo, y); //[krisskross]~ ERROR E0495
-   let b = bar(foo, x); //[krisskross]~ ERROR E0495
-   (a, b)
+   let a = bar(foo, y);
+   let b = bar(foo, x);
+   (a, b) //[krisskross]~ ERROR E0623
+   //[krisskross]~^ ERROR E0623
```

Comment:
Looks like one of the errors should point to line 64, instead of line 65.

```
[00:55:56] ---- [compile-fail] compile-fail/associated-types/cache/project-fn-ret-invariant.rs stdout ----
[00:55:56] 	
[00:55:56] error in revision `krisskross`: /checkout/src/test/compile-fail/associated-types/cache/project-fn-ret-invariant.rs:64: unexpected "error": '64:21: 64:22: lifetime mismatch [E0623]'
[00:55:56] 
[00:55:56] error in revision `krisskross`: /checkout/src/test/compile-fail/associated-types/cache/project-fn-ret-invariant.rs:65: expected error not found: E0623
[00:55:56] 
[00:55:56] error in revision `krisskross`: 1 unexpected errors found, 1 expected errors not found
[00:55:56] status: exit code: 101
[00:55:56] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/compile-fail/associated-types/cache/project-fn-ret-invariant.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/compile-fail" "--target=i686-unknown-linux-musl" "--cfg" "krisskross" "--error-format" "json" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/compile-fail/associated-types/cache/project-fn-ret-invariant.stage2-i686-unknown-linux-musl" "-Clinker=/musl-i686/bin/musl-gcc" "-Crpath" "-O" "-Lnative=/checkout/obj/build/i686-unknown-linux-musl/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/compile-fail/associated-types/cache/project-fn-ret-invariant.stage2-i686-unknown-linux-musl.compile-fail.libaux" "-A" "unused"
[00:55:56] unexpected errors (from JSON output): [
[00:55:56]     Error {
[00:55:56]         line_num: 64,
[00:55:56]         kind: Some(
[00:55:56]             Error
[00:55:56]         ),
[00:55:56]         msg: "64:21: 64:22: lifetime mismatch [E0623]"
[00:55:56]     }
[00:55:56] ]
[00:55:56] 
[00:55:56] not found errors (from test file): [
[00:55:56]     Error {
[00:55:56]         line_num: 65,
[00:55:56]         kind: Some(
[00:55:56]             Error
[00:55:56]         ),
[00:55:56]         msg: "E0623"
[00:55:56]     }
[00:55:56] ]
[00:55:56] 
[00:55:56] thread '[compile-fail] compile-fail/associated-types/cache/project-fn-ret-invariant.rs' panicked at 'explicit panic', /checkout/src/tools/compiletest/src/runtest.rs:1084:12
[00:55:56] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:55:56] 
[00:55:56] 
[00:55:56] failures:
[00:55:56]     [compile-fail] compile-fail/associated-types/cache/project-fn-ret-invariant.rs
[00:55:56] 
[00:55:56] test result: [31mFAILED(B[m. 2759 passed; 1 failed; 15 ignored; 0 measured; 0 filtered out
```

Reply:
As I feared..

Reply:
There's an unexpected randomness within the error span logic. Trying to find it now.